dapr/proto/runtime/v1/dapr.proto

```diff
@@ -1135,7 +1135,7 @@ message Job {
   string schedule = 2;
 
   // Job data.
-  google.protobuf.Any data = 3;
+  bytes data = 3;
```

Comment:
We want to keep this as `google.protobuf.Any`. Using `bytes` means that data will need to be unnecessarily marshaled twice- both when writing structured data to the object, and again when reading that data back. Using `Any` means the client provided data is stored as-is with its type kept intact, improving performance and space.

Clients who wish to store non-structured blob data can use the `google.protobuf.BytesValue` protobuf type.

Reply:
`data` is the customer's payload and we don't want to mix the serialization/deserialization of our request/response object with customer's payload. These bytes will be stored in etcd or state store and it is customer's app (or Dapr SDK) to serialize and deserialize their object. If the customer is sending a protobuf serialized value, it does matter from here on.

Reply:
This is the purpose of using the `Any` type here- we don't _need_ to do any serialisation on the client provided data, it is stored as is with its type definition kept intact. By enforcing a bytes type, we force the client to serialise their data into a byte slice and remove the proto type definition- the end result is clients marshalling and marshalling json which we want to avoid.

By keeping `Any` we retain proto type definition and prevent data marshalling which consumes CPU cycles.

https://protobuf.dev/programming-guides/proto3/#any

Reply:
How will the proto definition be kept in etcd or Redis? At some point, before writing to the database, we need bytes. The creation of bytes can happen at the customer's app or in the runtime. We are not in the business of dictating how customer decide to represent their data - it can be protbuf, JSON, XML, or anything they want. The common denominator is byte array.

Reply:
The proto `data` will be stored in the state store in the protobuf wire format, generally via https://github.com/golang/protobuf/blob/5d5e8c018a13017f9d5b8bf4fad64aaa42a87308/proto/wire.go#L22
The protobuf wire format keeps type integrity.

Indeed clients can decide on their own data formats however Dapr has standardized on protobuf, and their requests will always be encoded into protobuf [even when using HTTP](https://github.com/cicoyle/dapr/blob/c1b7c0f8decac190e514063d64538e3f5fd690a4/pkg/grpc/universalapi/api_scheduler.go#L33). If clients use gRPC (which is always preferred) they will see performance benefits where structured data doesn't need to be transformed on writes and reads. Clients can continue to use their own formats, however serialization needs to be done on the client side (as this PR forces for also gRPC).

Reply:
We have not standardized on Protobuf. We have a mix of JSON and protobuf. Let's discuss this outside the PR first and come back with a decision.

Reply:
I think both are viable choices but would prefer to go for protobuf in this case, as we use it for mainline data handling in core cases in Dapr (service invocation, actors) and the scheduler needs to optimize for performance which is where `Any` would likely give better results.

Reply:
I am changing this to be Any for now. It will also work if we decide we can just send byte[].